[WaitForARMFeedback] Long Term Retention for Managed Instances

xaliciayang opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

xaliciayang committed:
update swagger to accomodate for new managed db create mode (ltr)

AutorestCI commented:
You don't have permission to trigger SDK Automation.
Please add yourself to ***Azure*** group from opensource portal if you are MSFT employee,
or please ask reviewer to add comment *** /openapibot sdkautomation ***.
Please ask tih@microsoft.com (or NullMDR in github) for additional help.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=8266) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=8266) before we can accept your contribution.<br/><br/>:x: xaliciayang [sign now](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=8266)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-rest-api-specs?pullRequest=8266) it.</sub>

xaliciayang committed:
Adding example file for create managed db from ltr restore

xaliciayang committed:
remove unused example

AutorestCI commented:
/azp run automation - sdk

pixia commented on specification/sql/resource-manager/Microsoft.Sql/preview/2017-03-01-preview/managedDatabases.json:
```diff
@@ -16,64 +16,6 @@
     "application/json"
   ],
   "paths": {
-    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/locations/{locationName}/managedDatabaseRestoreAzureAsyncOperation/{operationId}/completeRestore": {
```
why is this included?

xaliciayang replied:
This is removed

xaliciayang committed:
undoing removal of completeRestore

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
adding back complete managed restore example

xaliciayang committed:
add examplee and swagger files for ltr backup/policies

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
update readme; add in 2018 manage databases diffs

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
remove create example from 2018

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
add -preview to swagger files added to readme

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
removing empty body from delete backup examples

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
fix parameter name in exmple file

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
add 2018 example. fix version in 2017 example

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
../../2018-06-01-preview/managedDatabases.json

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
removing example from 2018

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
ran prettier on some example files

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
adding other files called out by prettier

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
update readme to include swagger files in v3/v2/v1

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
Revert "update readme to include swagger files in v3/v2/v1"
Removing files committed by prettier-fix

This reverts commit db6d1464a3c4360fa29f331ad39a50859686b4e7.

xaliciayang committed:
reverted all unrelated files
updated readme to include swagger files in v3/v2/v1

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
update swagger after introducing new error handler

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
removing old versions of manageDatabase.json swagger from readme.md package composites (2018-06-01 version is referenced)

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented on specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json:
```diff
@@ -638,6 +639,13 @@
           "x-ms-mutability": [
             "create"
           ]
+        },
```
Is there a reason that 2019-06-01-preview version does not have this and LongTermRetentionManagedInstanceBackups.json?

xaliciayang replied:
Added managedDatabases.json to 2019-06-01-preview as well.  the Backups/Policies APIs are new and we are going to use 2018-06-01-preview for those.

xaliciayang committed:
Adding updated swagger files for:
managedDatabases
LongTermRetentionManagedInstanceBackups
ManagedInstanceLongTermRetentionPolicies

as well as relevant examples, to 2018 and 2019

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
add files after running prettier-fix; removing LTR backups/policies APIs from 2019

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
updated readme to use 2019 version of managedDatabases.json

AutorestCI commented:
/azp run automation - sdk

allenjzhang commented:
/azp run automation - sdk

allenjzhang commented:
Code style issues has been found in PR. Please follow [this link](https://aka.ms/AA6h31t) to fix the issues. Thanks!

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
adding files fixed by prettier

AutorestCI commented:
/azp run automation - sdk

ravbhatnagar commented on specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json:
```diff
@@ -0,0 +1,920 @@
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Sql/locations/{locationName}/longTermRetentionManagedInstances/{managedInstanceName}/longTermRetentionDatabases/{databaseName}/longTermRetentionManagedInstanceBackups/{backupName}": {
```
can these backups exist at both scopes - RG and subscription? can they be created at both these scopes? Why is the support needed for subscription scope? To use these, the caller will need /write /read permissions at the scope of a subscription which is more privileged than getting the permissions at a RG scope. 
A regular database is created under an RG.

xaliciayang replied:
Backups can exist with or without a RG. 

In cases of LTR backups tied to dropped instances, the resource will not have an RG: subscriptions/$subscriptionId/providers/Microsoft.Sql/Locations/$location/longTermRetentionManagedInstances/$instanceName/longTermRetentionDatabases/$databaseName/longTermRetentionManagedInstanceBackups/$backupName. 

However, backups belonging to live instances _will_ contain a RG.

ravbhatnagar commented on specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/LongTermRetentionManagedInstanceBackups.json:
```diff
@@ -0,0 +1,920 @@
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
```
WHat about the PUT api for this? how are these created?

xaliciayang replied:
These are created through the currently existing backup infrastructure.  Generally, full database backups are taken 1x/week and our workflow determines whether or not a full backup for a server/managed instance should be saved for long term retention.

xaliciayang replied:
In other words, a user cannot "create" a new backup

ravbhatnagar commented:
both the questions were addressed. Signing off from ARM side.

AutorestCI commented:
/azp run automation - sdk

xaliciayang committed:
prettierfix.diff on LTR MI example

AutorestCI commented:
/azp run automation - sdk
